src/_pytest/config/__init__.py

```diff
@@ -1305,6 +1316,69 @@ def _getconftest_pathlist(
             values.append(relroot)
         return values
 
+    def _get_append_ini_values(self, name: str, ini_type) -> List[str]:
+        value = []
+        # append_ini is a list of "ini=value" options
+        # append all values if multiple values are set for same ini-name,
+        # e.g. -a foo=bar1 -a foo=bar2 will append 'bar1 bar2' to foo
+        for ini_config in self._append_ini:
+            try:
+                key, append_ini_value = ini_config.split("=", 1)
+            except ValueError:
+                raise UsageError(
+                    "-a/--append-ini expects option=value style (got: {!r}).".format(
+                        ini_config
+                    )
+                )
+            else:
+                if key == name:
+                    if ini_type in ["pathlist", "args"]:
+                        value.append(append_ini_value)
+                    elif ini_type == "linelist":
+                        value.append("\n{}".format(append_ini_value))
```

Comment:
This one looks a bit strange, are you sure it's correct? I haven't checked but I'd expect one of two things:

1. Single string separated by newlines: `'value1\nvalue2\nvalue3'`
2. List of strings without newlines: `['value1', 'value2', 'value3']`

But this creates

3. `['\nvalue1', '\nvalue2', '\nvalue3']`

Reply:
I did a check when I implemented this, and iirc it was correct

Reply:
If I resolved in error, please reopen